### ISSUE TITLE
browserTitle&title - not taking dynamic values when run in a loop with multiple collections

### ISSUE THREAD

**Issue by bhaskarnimma** [DannyDainton/newman-reporter-htmlextra#205]
**Describe The Problem**

I am running multiple collections in a loop and generating the reports into multiple files. Report file names are working correctly. But the title and browser title is taken from the last element of the array. 

**Configuration**

```
let collections = ["collection_1", "collection_2"];
for(var i = 0; i< collections.length; i++){
   newman.run({
        collection: require(`./postman/${collections[i].json}`),
        insecure: true,
        reporters: ['cli', 'htmlextra'],
        reporter: {
            htmlextra: {
                browserTitle: `${collections[i]} - Automation Summary Report`,
                title: `${collections[i]} - Automation Summary  Report`,
                logs: "true",
                export: `./results/${collections[i]-report.html}`
            }
        }
    }, function (err) {
        if (err) { throw err; }
        console.log('\n------------------------------------------- \n')
        console.log(`${collection.name} run complete! check the results folder`);
    });
}
```
 

**What Are You Seeing**

When running with multiple collections in a loop, the report title is taken from the last element of the array. 

`let collections = ["collection_1.json", "collection_2.json"];`

it is generating 2 reports `collection_1-report.html` & `collection_2-report.html`.
But the report "collection_1-report.html"'s `browserTitle` and `title` are displaying as `collection_2 - Automation Summary  Report`. It should be  `collection_1 - Automation Summary  Report`

**Version and Platform Information:**

 - "newman": "^5.0.0",
 -  "newman-reporter-htmlextra": "^1.14.1"
 - OS [Windows 10]

**Comment by DannyDainton** [comment]
Hey @bhaskarnimma,

Let me take a look and see what's going on there. Looking at your script, it looks like it has a few issues with the `${collection[i]}` references to get the collection and to create the report. 

I think it should be like this:

let collections = ["collection_1", "collection_2"];

```
for(var i = 0; i < collections.length; i++){
   newman.run({
        collection: require(`./postman/${collections[i]}.json`),
        //insecure: true,
        reporters: ['cli', 'htmlextra'],
        reporter: {
            htmlextra: {
                browserTitle: `${collections[i]} - Automation Summary Report`,
                title: `${collections[i]} - Automation Summary Report`,
                //logs: "true",
                export: `./results/${collections[i]}-report.html`
            }
        }
    }, function (err, summary) {
        if (err) { throw err; }
        console.log('\n------------------------------------------- \n')
        console.log(`${summary.collection.name} run complete! check the results folder`);
    });
}
```

I can report the issue you're facing but I need to figure out why it's creating the reports with that title. Apart from the title, are the reports the same?

**Comment by DannyDainton** [comment]
There's a couple of helpers that decide what goes in there based on the options given, I added a log statement to see what it populates at runtime:

![Screenshot 2020-05-05 at 12 33 29](https://user-images.githubusercontent.com/17089546/81061675-af13bb80-8ecc-11ea-8632-a4c46df6e3ae.png)

As you can see, that part of the script is looping through that first and for some reason picking the last item in the array and using that in the reports before the other part of the script has run.

![Screenshot 2020-05-05 at 12 33 18](https://user-images.githubusercontent.com/17089546/81061949-26494f80-8ecd-11ea-95d8-3ef4fb5b97ae.png)

**Comment by DannyDainton** [comment]
@bhaskarnimma  Fixed the issue locally, when I get time tomorrow I will push the changes :) 

![Screenshot 2020-05-05 at 19 22 57](https://user-images.githubusercontent.com/17089546/81101345-e3579e00-8f05-11ea-9142-e4d912e707e9.png)

**Comment by DannyDainton** [comment]
@bhaskarnimma If you update to the new version, this should be working now - Let me know if it's not. :)

**Comment by bhaskarnimma** [comment]
Hi @DannyDainton, I tested the fix with latest version (1.15.0) and it is working as expected. Thank you so much for quick turn around.

**Comment by DannyDainton** [comment]
Awesome! Thanks for reporting that and glad to hear it's working for you now. 🏆
